src/Plugin_Command.php

```diff
@@ -1148,13 +1152,35 @@ public function uninstall( $args, $assoc_args = array() ) {
 				}
 			}
 
-			if ( ! Utils\get_flag_value( $assoc_args, 'skip-delete' ) && $this->delete_plugin( $plugin ) ) {
-				WP_CLI::log( "Uninstalled and deleted '$plugin->name' plugin." );
+			if ( ! Utils\get_flag_value( $assoc_args, 'skip-delete' ) ) {
+				if ( $this->delete_plugin( $plugin ) ) {
+					$deleted_plugin_files[] = $plugin->file;
+					WP_CLI::log( "Uninstalled and deleted '$plugin->name' plugin." );
+				} else {
+					$delete_errors[] = $plugin->file;
+					WP_CLI::log( "Ran uninstall procedure for '$plugin->name' plugin. Deletion failed" );
+					++$errors;
+					continue;
+				}
 			} else {
 				WP_CLI::log( "Ran uninstall procedure for '$plugin->name' plugin without deleting." );
 			}
 			++$successes;
 		}
+
+		// Remove deleted plugins from the plugin updates list.
```

Comment:
the assumption here is that only deleted plugins should be removed as the others still have files around? And as a result it would be good to keep those files up-to-date